CHANGELOG.md

```diff
@@ -30,23 +30,33 @@ _None._
 
 -->
 
-## 4.1.0-beta.1 [Unreleased]
+## 4.2.0 [Unreleased]
```

Comment:
The following is out of scope from this PR, because it correctly update the code based on its existing state, and also a big nitpick.

IMHO, this header should always stay "Unreleased" because it's up to the release manager (eventually the automation) to parse what went into the changelog since the latest release and decide what number to use based on the kind of changes.

I appreciate the idea of suggesting a new version number but I feel it would be neater if the header for unreleased code stayed fixed.

I might open a PR later on to update the template to make this clearer and we can discuss then.

Reply:
Agreed. Predicted next version number will vary as new items will get added to the unreleased section, which is why we don't include a version number in it until we actually do the next release and decide on it based on content of the section and semver rules